### DISCUSSION TITLE
Is there an efficient way to measure the memory usage?

### DISCUSSION THREAD

**Discussion by PenutChen** [huggingface/text-generation-inference#1157] [Q&A] [answered]
I'm measuring the relationship between sequence length, prefilled batch size, and GPU memory usage. For now, the only way I can measure memory usage is through trial and error using the `--cuda-memory-fraction` flag, since TGI seems to use all the available memory by default (I think this is caused by the paged attention mechanism?).

I'm wondering if there's a more efficient way to do this?

**Comment by OlivierDehaene** [comment, marked as answer]
Not really no. You could add a few prints [here](https://github.com/huggingface/text-generation-inference/blob/7402a355dcbf9ffe7a0b2a788f2062aa9e0a3ed5/server/text_generation_server/models/flash_causal_lm.py#L687) to get the info you need.

**Reply by PenutChen** [reply]
Thanks, I'll try to dig into it!
